tools/src/main/java/org/neo4j/tools/txlog/checktypes/RelationshipCheckType.java

```diff
@@ -43,14 +46,35 @@ public RelationshipRecord after( Command.RelationshipCommand command )
         return command.getAfter();
     }
 
+    /**
+     * Checks that two given {@link RelationshipRecord}s are equal. {@link RelationshipRecord#inUse() Used}
+     * records are compared by all present fields/pointers. Unused records are compared only by id. This is so
+     * because for removed relationships we write not only inUse flag but also
+     * {@link RelationshipRecord#getType() relationship type} in
+     * {@link Command.RelationshipCommand#serialize(WritableChannel)} and read it in
+     * {@link PhysicalLogCommandReaderV3_0_2#readRelationshipRecord(long, ReadableChannel)}.
+     *
+     * @param record1 first record to check.
+     * @param record2 second record to check.
+     * @return {@code true} when records are equal, otherwise {@code false}.
+     */
     @Override
     public boolean equal( RelationshipRecord record1, RelationshipRecord record2 )
     {
         Objects.requireNonNull( record1 );
         Objects.requireNonNull( record2 );
 
-        return record1.getId() == record2.getId() &&
-               record1.inUse() == record2.inUse() &&
+        if ( record1.getId() != record2.getId() )
+        {
+            return false;
+        }
+
+        if ( record1.inUse() == record2.inUse() && !record1.inUse() )
```

Comment:
This looks like more like a general rule: we need to check only ids in not used records for equality checks; maybe we need to add similar condition to `NodeCheckType` and `RelationshipGroupCheckType` (`PropertyCheckType` already have some for of this check)?